pkg/kubernetes/kubernetes.go

```diff
@@ -26,23 +29,51 @@ const (
 
 type CloseWatchKubeConfig func() error
 
-type Kubernetes struct {
+type Kubernetes interface {
+	WatchKubeConfig(onKubeConfigChange func() error)
+	Close()
+	Derived(ctx context.Context) DerivedKubernetes
+	ConfigurationView(minify bool) (runtime.Object, error)
+	IsOpenShift(ctx context.Context) bool
+}
+
+type DerivedKubernetes interface {
+	IsOpenShift(ctx context.Context) bool
+	CacheInvalidate()
+	NewHelm() *helm.Helm
+	EventsList(ctx context.Context, namespace string) ([]map[string]any, error)
+	NamespacesList(ctx context.Context, options ResourceListOptions) (runtime.Unstructured, error)
+	PodsListInAllNamespaces(ctx context.Context, options ResourceListOptions) (runtime.Unstructured, error)
+	PodsListInNamespace(ctx context.Context, namespace string, options ResourceListOptions) (runtime.Unstructured, error)
+	PodsGet(ctx context.Context, namespace, name string) (*unstructured.Unstructured, error)
+	PodsDelete(ctx context.Context, namespace, name string) (string, error)
+	PodsLog(ctx context.Context, namespace, name, container string) (string, error)
+	PodsRun(ctx context.Context, namespace, name, image string, port int32) ([]*unstructured.Unstructured, error)
+	PodsTop(ctx context.Context, options PodsTopOptions) (*metrics.PodMetricsList, error)
+	PodsExec(ctx context.Context, namespace, name, container string, command []string) (string, error)
+	ProjectsList(ctx context.Context, options ResourceListOptions) (runtime.Unstructured, error)
+	ResourcesList(ctx context.Context, gvk *schema.GroupVersionKind, namespace string, options ResourceListOptions) (runtime.Unstructured, error)
+	ResourcesGet(ctx context.Context, gvk *schema.GroupVersionKind, namespace, name string) (*unstructured.Unstructured, error)
+	ResourcesCreateOrUpdate(ctx context.Context, resource string) ([]*unstructured.Unstructured, error)
+	ResourcesDelete(ctx context.Context, gvk *schema.GroupVersionKind, namespace, name string) error
+}
+
+type kubernetes struct {
```

Comment:
I'd recommend renaming it to `KubernetesManager`. I don't think, exporting the fields in here hurts us. However, in my opinion, maybe we can get rid of some of them (e.g. I bet we don't need parameterCodec by using an upstream functionality.).